rllib/algorithms/algorithm.py

```diff
@@ -2646,11 +2678,15 @@ def _checkpoint_info_to_algorithm_state(
                 if pid in policy_ids
             }
 
+            # Get Algorithm class.
+            if isinstance(state["algorithm_class"], str):
+                # Try deserializing from a full classpath.
+                # Or as a last resort: Tune registered algorithm name.
+                state["algorithm_class"] = deserialize_type(
```

Comment:
note to self: check if deserialize_type returns none if type doesn't exist, or raises an error.